assembl/static2/js/app/components/svg/doughnut.jsx

```diff
@@ -1,36 +1,96 @@
 import React from 'react';
+import { OverlayTrigger } from 'react-bootstrap';
+
 import { calculatePercentage } from '../../utils/globalFunctions';
 
-export default ({ like, disagree }) => {
-  const totalCount = like + disagree;
-  const radius = 2.5;
-  const circumference = 2 * radius * Math.PI;
-  const elements = [{ name: 'green', count: like }, { name: 'red', count: disagree }];
-  let lastSize = 0;
-  return (
-    <svg className="doughnut" height="10em" width="10em">
-      {elements
-        .filter(({ count }) => {
-          return count > 0;
-        })
-        .map(({ name, count }, index) => {
-          const normalizedSize = calculatePercentage(count, totalCount) / 100;
-          const offset = lastSize;
-          const size = circumference * normalizedSize;
-          lastSize = size;
-          const gap = circumference - size;
-          return (
-            <circle
-              key={index}
-              className={`circle radial-${name}`}
-              r={`${radius}em`}
-              cx={`${5}em`}
-              cy={`${5}em`}
-              strokeDasharray={`${size}em ${gap}em`}
-              strokeDashoffset={`${offset}em`}
-            />
-          );
-        })}
+function polarToCartesian(centerX, centerY, radius, angleInDegrees) {
+  const angleInRadians = (angleInDegrees - 90) * Math.PI / 180.0;
+
+  return {
+    x: centerX + radius * Math.cos(angleInRadians),
+    y: centerY + radius * Math.sin(angleInRadians)
+  };
+}
+
+const describeArc = (x, y, radius, startAngle, endAngle) => {
+  const start = polarToCartesian(x, y, radius, endAngle);
+  const end = polarToCartesian(x, y, radius, startAngle);
+
+  const largeArcFlag = endAngle - startAngle <= 180 ? '0' : '1';
+
+  const d = ['M', start.x, start.y, 'A', radius, radius, 0, largeArcFlag, 0, end.x, end.y].join(' ');
+
+  return d;
+};
+
+const cx = 70;
+const cy = 70;
+const r = 50;
+
+const getColor = (element) => {
+  return 'color' in element ? element.color : 'lightgrey';
+};
+
+const simpleCircle = (element) => {
+  return 'Tooltip' in element
+    ? <OverlayTrigger container={this} overlay={element.Tooltip} placement="bottom">
+      <circle className="circle" cx={cx} cy={cy} r={r} stroke={getColor(element)} />
+    </OverlayTrigger>
+    : <circle className="circle" cx={cx} cy={cy} r={r} stroke={getColor(element)} />;
+};
+
+const placementFromAngle = (angle) => {
+  const firstQuartant = 360 - 45;
```

Comment:
Comment where each quadrant is. Seems that this definition of quadrants is different than what is taught in Canadian school system. Eg. ```// bottom right```

Reply:
oh, quartant is not en english word xD